src/services/destination/nativeIntegration.ts

```diff
@@ -191,24 +191,22 @@ export class NativeIntegrationDestinationService implements DestinationService {
       const processedProxyResponse = networkHandler.processAxiosResponse(rawProxyResponse);
       let rudderJobMetadata =
         version.toLowerCase() === 'v1'
-          ? (deliveryRequest as ProxyDeliveriesRequest).metadata
-          : (deliveryRequest as ProxyDeliveryRequest).metadata;
+          ? (deliveryRequest as ProxyV1Request).metadata
+          : (deliveryRequest as ProxyV0Request).metadata;
 
       if (version.toLowerCase() === 'v1' && handlerVersion.toLowerCase() === 'v0') {
         rudderJobMetadata = rudderJobMetadata[0];
       }
-
-      let responseProxy = networkHandler.responseHandler(
-        {
-          ...processedProxyResponse,
-          rudderJobMetadata,
-        },
-        destinationType,
-      );
+      const responseParams = {
```

Comment:
This is the core part where the change has been done which affects all network handlers